test/broadcast.jl

```diff
@@ -282,3 +282,30 @@ end
     end
 
 end
+
+# A help struct to test style-based broadcast dispatch with unknown array wrapper.
+# `WrapArray(A)` behaves like `A` during broadcast. But its not a `StaticArray`.
+struct WrapArray{T,N,P<:AbstractArray{T,N}} <: AbstractArray{T,N}
+    data::P
+end
+Base.@propagate_inbounds Base.getindex(A::WrapArray, i::Integer...) = A.data[i...]
+Base.@propagate_inbounds Base.setindex!(A::WrapArray, v::Any, i::Integer...) = setindex!(A.data, v, i...)
+Base.size(A::WrapArray) = size(A.data)
+Broadcast.BroadcastStyle(::Type{WrapArray{T,N,P}}) where {T,N,P} = Broadcast.BroadcastStyle(P)
+StaticArrays.isstatic(A::WrapArray) = StaticArrays.isstatic(A.data)
+StaticArrays.Size(::Type{WrapArray{T,N,P}}) where {T,N,P} = StaticArrays.Size(P)
+function StaticArrays.similar_type(::Type{WrapArray{T,N,P}}, ::Type{t}, s::Size{S}) where {T,N,P,t,S}
+    return StaticArrays.similar_type(P, t, s)
+end
+
+@testset "Broadcast with unknown wrapper" begin
```

Comment:
Could you add a test where the first array is a transposed static array? I think something like `@test @inferred(transpose(b) .+ transpose(a)) isa T` would be fine.

You could also bump patch version, then I'll merge this and tag a new release.

Reply:
`transpose(b) .+ transpose(a)` wont be a `StaticArray`, as `b` is not a `StaticArray`.
If we want to propagate `BroadcastStyle` more eagerly, we'd better change this behaviour in `LinearAlgebra`.

Reply:
Ah, right, we can't transpose `b` here. Let's leave this as is then.